TACC-Cloud/hazmapper
# task/WG-217: add features requests for map

## Overview: ##

Add features requests to be used in MapProject

## PR Status: ##


## Related Jira tickets: ##

* [WG-217](https://tacc-main.atlassian.net/browse/WG-217)

## Summary of Changes: ##

- Add @types/geojson and improve the features types
- Improve support for public routes of projects
- Added userFeature hook
- Add turf library and use to zoom to extend of loaded features
- Add query parameters used in the projects/xxx/features route as the Analytics team uses that information
- Fixed an issue with login/logout and ensure react-query caches are cleared
- Refactor auth state from `token.token` to `authToken.token` to be less error prone for developers.

## Testing Steps: ##

1. Run local geoapi for angular/react hazmapper (ensure you are using local geoapi in react/src/secret_local.ts and in and in src/environments/environment.ts)
2. Use **angular** hazmapper to make a map and add some features (note the route to your map, e.g., http://localhost:4200/project/SOME_UUID).  **Also make the map public.**
3. Use **react** hazmapper and go to the same map route.
4. Ensure that features are loaded
5. Ensure that the logging statement looks correct and has app:hazmapper (i.e. INFO :: [projects.py:309] :: Get features of project for user:nathanf application:Unknown public_view:True project_uuid:4dd7f996-728f-4c78-9f6d-ad61f1c768b9 project:31 tapis_system_id:designsafe.storage.default tapis_system_path:/nathanf )
6. You can also view public map (i.e. http://localhost:4200/public-project/SOME_UUID)
7. Logout by http://localhost:4200/logout and then view public map again as a guest by http:/localhost:4200/logout and then http://localhost:4200/public-project/SOME_UUID.   Check if log statements make sense.
